scripts/s3uploading.js

```diff
@@ -0,0 +1,149 @@
+const shell = require("shelljs");
+const AWS = require('aws-sdk');
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+
+const workdir = process.env.GITHUB_WORKSPACE;
+const s3 = new AWS.S3({
+    "AWS_ACCESS_KEY_ID": process.env.AWS_ACCESS_KEY_ID,
+    "AWS_SECRET_ACCESS_KEY": process.env.AWS_SECRET_ACCESS_KEY
+});
+const tmp = `${workdir}/tmp`;
+const readmeRegex = new RegExp('^readme((\.(org|md|rst)$)|$)', 'i'); // readme, readme.org, readme.md, readme.rst
+const Bucket = process.env.BUCKET_NAME
+
+const readJSON = (file) => JSON.parse(fs.readFileSync(file, 'utf8'));
+
+const modules = readJSON(`${workdir}/index.json`).modules;
+const versions = readJSON(`${workdir}/versions.json`);
+
+let commitMsg = ["Updated versions.json\n"];
+
+let uploadQueue = [];
+
+const exit = err => {
+    console.error(`\x1b[31m${err}\x1b[0m`);
+    process.exit(1);
+}
+
+const uploadFile = (file, s3path) => s3.upload({
+    Bucket,
+    Key: s3path,
+    Body: fs.readFileSync(file),
+    ACL: 'public-read'
+}, (err, data) => {
+    if (err) exit(`Error while uploading. Err.: ${err}`)
+    if (data) console.log(`${data.Location} successfully uploaded`);
+})
+
+const createTMP = () => {
+    shell.rm('-rf', tmp);
+    shell.mkdir(tmp);
+}
+createTMP();
```

Comment:
Why call the function here instead of inside the `try` below?

Reply:
done